tool/tsh/tctl_test.go

```diff
@@ -94,3 +96,81 @@ func TestLoadConfigFromProfile(t *testing.T) {
 		})
 	}
 }
+
+func TestSetAuthServerFlagWhileLoggedIn(t *testing.T) {
+	tmpHomePath := filepath.Clean(t.TempDir())
+	connector := mockConnector(t)
+
+	alice, err := types.NewUser("alice@example.com")
+	require.NoError(t, err)
+	alice.SetRoles([]string{"access"})
+
+	authProcess, proxyProcess := makeTestServers(t, withBootstrap(connector, alice))
+
+	authServer := authProcess.GetAuthServer()
+	require.NotNil(t, authServer)
+
+	authAddr, err := authProcess.AuthAddr()
+	require.NoError(t, err)
+
+	proxyAddr, err := proxyProcess.ProxyWebAddr()
+	require.NoError(t, err)
+
+	err = Run(context.Background(), []string{
+		"login",
+		"--insecure",
+		"--debug",
+		"--auth", connector.GetName(),
+		"--proxy", proxyAddr.String(),
+	}, setHomePath(tmpHomePath), cliOption(func(cf *CLIConf) error {
+		cf.mockSSOLogin = mockSSOLogin(t, authServer, alice)
+		return nil
+	}))
+	require.NoError(t, err)
+	// we're now logged in with a profile in tmpHomePath.
+
+	tests := []struct {
+		desc           string
+		authServerFlag []string
+		want           []utils.NetAddr
+	}{
+		{
+			desc: "sets default web proxy addr without auth server flag",
+			want: []utils.NetAddr{*proxyAddr},
+		},
+		{
+			desc:           "sets auth addr from auth server flag ignoring profile setting",
+			authServerFlag: []string{authAddr.String()},
+			want:           []utils.NetAddr{*authAddr},
+		},
+		{
+			desc:           "sets auth addr from auth server flag when profile is not found",
+			authServerFlag: []string{"site.not.in.profile.com:3080"},
+			want: []utils.NetAddr{
+				{
+					Addr:        "site.not.in.profile.com:3080",
+					AddrNetwork: "tcp",
+					Path:        "",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			ccf := new(common.GlobalCLIFlags)
+			ccf.AuthServerAddr = tt.authServerFlag
+
+			cfg := new(service.Config)
+			cfg.TeleportHome = tmpHomePath
+			// this is needed for the case where the --auth-server=host is not found in profile.
+			// ApplyConfig will try to read local auth server identity if the profile is not found.
+			cfg.DataDir = authProcess.Config.DataDir
+
+			_, err = common.ApplyConfig(ccf, cfg)
+			require.NoError(t, err)
+			require.NotEmpty(t, cfg.AuthServers)
+
+			require.ElementsMatch(t, tt.want, cfg.AuthServers)
```

Comment:
Suggestion: add explicit error messages to those requires, since this is the method/failures we are most interested in?

Reply:
I'll add a message to the require.NotEmpty, since that's the only I think isn't self-explanatory from the test's description:

```go
require.NotEmpty(t, cfg.AuthServers, "auth servers should be set to a non-empty default if not specified")
```